### ISSUE TITLE
Fail to init Meross Lan at HA startup or when trying to setup manually

### ISSUE THREAD

**Issue by mfrederic** [krahabb/meross_lan#311]
## Version of the custom_component

Meross Lan v4.3.0

## Configuration

__Hardware__: Raspberry 4 model b

Running on Raspbian v11 (bullseye)

Running HA Core with version 2023.1.7

__No successful running of Meross LAN__ so no configuration related to it

## Describe the bug
The Meross Lan cannot initialize at HA startup or when intenting to set it up manually.

Both trigger the same error that can be seen below in the debug log section.

The manual setup triggers an extra error log also added below. It also shows that error message:

<img width="530" alt="image" src="https://github.com/krahabb/meross_lan/assets/3066614/0cd99f2f-e033-46bb-bce7-904c64b4d046">

## Debug log

```text
This error originated from a custom integration.

Logger: homeassistant.loader
Source: custom_components/meross_lan/const.py:27
Integration: Meross LAN (documentation, issues)
First occurred: 09:38:05 (2 occurrences)
Last logged: 09:42:13

Unexpected exception importing platform custom_components.meross_lan.config_flow
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/loader.py", line 744, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/loader.py", line 761, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/homeassistant/.homeassistant/custom_components/meross_lan/__init__.py", line 15, in <module>
    from .const import (
  File "/home/homeassistant/.homeassistant/custom_components/meross_lan/const.py", line 27, in <module>
    CONF_PROTOCOL_OPTIONS: dict[str | None, str] = {
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'

```

When trying to manually setup, also this is logged:

```text
Logger: homeassistant.config_entries
Source: config_entries.py:852
First occurred: 09:38:05 (2 occurrences)
Last logged: 09:42:13

Error occurred loading configuration flow for integration meross_lan: Exception importing custom_components.meross_lan.config_flow
```

**Comment by dabbler68** [comment]
Same problem here since updating HA OS to 10.4 and core to 2023.8.3 (from 10.3 and 2023.7.3). 
Generic on Intel.

None of my Meross devices are visible, but they all work via my Meross phone app.

Hard reboot again just now and these are the logfile entries:

Traceback (most recent call last):
  File "/config/custom_components/meross_lan/helpers.py", line 586, in exception_warning
    yield
  File "/config/custom_components/meross_lan/meross_device.py", line 183, in __init__
    device_registry.async_get(ApiProfile.hass).async_get_or_create(
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 511, in async_get_or_create
    device_info_type = _validate_device_info(config_entry, device_info)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 165, in _validate_device_info
    raise DeviceInfoError(
homeassistant.helpers.device_registry.DeviceInfoError: Invalid device info {'connections': {('mac', '48:e1:xxxxx2f')}, 'default_manufacturer': 'Meross', 'default_name': 'Smart Light (211018137424xxxxxxxxxx2f)', 'identifiers': {('meross_lan', '211018137424xxxxxxxxxxx2f')}, 'model': 'msl430 4.0.0', 'sw_version': '4.2.6', 'via_device': None} for 'meross_lan' config entry: device info needs to either describe a device, link to existing device or provide extra information.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 388, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/meross_lan/__init__.py", line 516, in async_setup_entry
    device = api.build_device(config_entry)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/meross_lan/__init__.py", line 402, in build_device
    device = class_type(descriptor, config_entry)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/meross_lan/switch.py", line 50, in __init__
    super().__init__(descriptor, entry)
  File "/config/custom_components/meross_lan/light.py", line 434, in __init__
    super().__init__(descriptor, entry)
  File "/config/custom_components/meross_lan/meross_device.py", line 351, in __init__
    super().__init__(
  File "/config/custom_components/meross_lan/meross_device.py", line 181, in __init__
    with self.exception_warning("DeviceRegistry.async_get_or_create"):
  File "/usr/local/lib/python3.11/contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/config/custom_components/meross_lan/helpers.py", line 588, in exception_warning
    self.warning(
  File "/config/custom_components/meross_lan/meross_device.py", line 471, in warning
    if self._trace_file:
       ^^^^^^^^^^^^^^^^
AttributeError: 'ToggleXMixinLightMixinMerossDevice' object has no attribute '_trace_file'

**Comment by mfrederic** [comment]
Thanks for the feedback on your experience. I'm gonna check if updating HA to 2023.7.3 will do on my side too.

I'll keep you in the loop 👍

**Comment by krahabb** [comment]
These are 2 different issues:
- @dabbler68: the issue you are experiencing is due to  a compatibility problem between `meross_lan` and HA core 2023.8.x. This has been fixed in `meross_lan` 4.3.0 so you should be fine just updating `meross_lan` to the latest release
- @mfrederic: this kind of 'brute' errors while initializing are usually related to corrupted files in the component sources. If this is the case they're always fixed by fully redownloading `meross_lan`

**Comment by mfrederic** [comment]
@krahabb I intended to download again the lib from scratch but it didn't help

The solution to update HA that was proposed by @dabbler68 plus your message about the v4.3.0 being compatible with latest HA actually worked.

Now I can add all my smart objects and use them flawlessly.
Thanks for the support from both of you.
